### PR TITLE
ET-3581 Implementing the NLB module

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- What does the pull request add and/or solve?

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/modules/nlb/main.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  target_group_arn = aws_lb_target_group.target_group.arn
+  target_id        = var.alb_id
+  port             = var.listener_port
+}
+
+resource "aws_lb" "this" {
+  name               = "${var.name}-alb-load-balancer"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = var.subnets.*
```
Comment (Robert-Steiner) [inline aws/modules/nlb/main.tf:20]:
```suggestion
  subnets            = var.subnets
```

File: aws/modules/nlb/variables.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+variable "subnets" {
+  description = "The IDs of the subnets to associated with the ALB"
+  type        = list(string)
+}
+
+variable "listener_port" {
+  description = "Value of the port for the listener"
+  type        = number
+}
```
Comment (Robert-Steiner) [inline aws/modules/nlb/variables.tf:24]: tags are missing

File: aws/modules/nlb/main.tf [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+resource "aws_lb_listener" "listener" {
+  load_balancer_arn = aws_lb.this.id
+  port              = var.listener_port
+  protocol          = "TCP"
+
+  default_action {
+    target_group_arn = aws_lb_target_group.target_group.id
+    type             = "forward"
+  }
```
Comment (Robert-Steiner) [inline aws/modules/nlb/main.tf:32]:
```suggestion
  }
   tags        = var.tags
```

**Review by Robert-Steiner — COMMENTED** [review]
i noticed is that we are missing the required terraform and aws version. here is an [example](https://github.com/xaynetwork/xayn_infrastructure/pull/710/files#diff-be21d12daee6761ddcfab69ed0aa69a397e1e27380ad9c84e140d1a00c3c759e)
we also need to add them in the vpc and alb module

**Comment by jorge-cr-13** [comment]
> i noticed is that we are missing the required terraform and aws version. here is an [example](https://github.com/xaynetwork/xayn_infrastructure/pull/710/files#diff-be21d12daee6761ddcfab69ed0aa69a397e1e27380ad9c84e140d1a00c3c759e)
> we also need to add them in the vpc and alb module

Don't you think is better to put this file when the module is called? Especially if we are using terragrunt, because it creates it by itself.

**Comment by jorge-cr-13** [comment]
> > > i noticed is that we are missing the required terraform and aws version. here is an [example](https://github.com/xaynetwork/xayn_infrastructure/pull/710/files#diff-be21d12daee6761ddcfab69ed0aa69a397e1e27380ad9c84e140d1a00c3c759e)
> > > we also need to add them in the vpc and alb module
> > 
> > 
> > Don't you think is better to put this file when the module is called? Especially if we are using terragrunt, because it creates it by itself.
> 
> i think we will need this for each module. the terraform file in the module will make sure that it is only be called by terraform version that is compatible with the code inside the module.

We just have to make sure to not add the `provider` information afterwards, because this is not rewritable at least for modules.

**Review by Robert-Steiner — APPROVED** [review]

**Pull request merged**